### ISSUE TITLE
xcodebuild failing due to errors extracting binaries of FirebaseAnalytics, GoogleAppMeasurement targets

### ISSUE THREAD

**Issue by bdrelling** [firebase/firebase-ios-sdk#8719]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.0 (13A233)
  * Firebase SDK version: 8.7.0
  * Installation method: `Swift Package Manager`
  * Firebase Component: Analytics, GoogleAppMeasurement

### [REQUIRED] Step 2: Describe the problem

I am never able to build my app via `xcodebuild` (via `fastlane`) anymore, as while packages are being resolved, I get the following error:

```
xcodebuild: error: Could not resolve package dependencies:
  artifact of binary target 'GoogleAppMeasurementIdentitySupport' failed extraction: The operation couldn’t be completed. (TSCBasic.StringError error 1.)
  artifact of binary target 'FirebaseAnalytics' failed extraction: The operation couldn’t be completed. (TSCBasic.StringError error 1.)
  artifact of binary target 'GoogleAppMeasurement' failed extraction: The operation couldn’t be completed. (TSCBasic.StringError error 1.)
```

I have since removed `FirebaseAnalytics` from my app entirely (I am so tired of the `GoogleAppMeasurement` build failures 1/4th of the time I try to build via command line), and there is no change. I was hoping that in doing so, it would stop caring about these targets.

#### Steps to reproduce:

I'm just running `fastlane gym`, which outputs the following:

```sh
xcodebuild -resolvePackageDependencies -workspace MyWorkspace.xcworkspace -scheme MyScheme
```

If I run only the `xcodebuild` command, it's still a 50/50 shot on whether it fails.

**Comment by bdrelling** [comment]
Okay, in _this particular instance_, it may be that `xcodebuild` was using the `12.5.1` toolset. Re-running for `13.0` explicitly works. Or at least, it has been consistent the most recent three builds (which is atypically stable for the `GoogleAppMeasurement` target).

However, I still get failures around `GoogleAppMeasurement` _all the time_ and I'm unable to consistently reproduce or diagnose. Sorry if this isn't helpful, but hopefully at least signaling a report here might raise the chance of it getting looked at if others raise similar reports down the road as well.

Feel free to close this issue.

**Comment by ryanwilson** [comment]
Thanks for the report @bdrelling and sorry for the frustrating issue. We've seen similar things in the past, and I've raised them with the SwiftPM team but unfortunately we haven't been able to track it down. Best I can tell is this is the line where the issue is thrown (message has changed slightly recently): https://github.com/apple/swift-package-manager/blob/3d48f94174fe484afbb854abb806cac9a51634ed/Sources/Workspace/Workspace.swift#L1980-L1981

The archiver is a straightforward call out to the system's `zip` command, and for whatever reason that fails. I've tried unzipping on a loop 10000 times in the command line but have never had it fail for me once, so something else must be getting in the way.

I'll close this for now since there's not much we can take action on at the moment, but thank you for the report and we'll see if we can reproduce at some point in the near future.
